app/helpers/admin/site_maps_helper.rb

```diff
@@ -7,7 +7,7 @@ module Admin::SiteMapsHelper
   #
   # @!visibility private
   def sitemap_pages
-    @sitemap_pages ||= proper_pages.only(:id, :name, :page_id, :path).group_by(&:page_id)
+    @sitemap_pages ||= proper_pages.only(:id, :name, :page_id, :path, :position, :role).group_by(&:page_id)
```

Comment:
Ok but with that other `only` scope the theory is that older Mongoid used to include `_id` by default. That wouldn't be the case with these fields, so something else must have changed...